src/Expr.cpp

```diff
@@ -0,0 +1,12 @@
+#include "Expr.h"
+#include "IR.h"
+
+namespace Halide {
+// FIXME: This feels like a horrible hack. FloatImm should be able
```

Comment:
It's a bit of a hack, but I disagree with the comment. The options seem to be:

- Making FloatImm more general to serve a rare use-case, affecting every use of it
- Adding Float16Imm, affecting every visitor, albeit in a very minor way
- Use a cast of a float

Options 3 really seems the least bad to me, in the sense that it interacts with the rest of the system the least.

The bigger wart is that we have no way to represent double immediates without losing precision.